### PR TITLE
feat(Makefile): enable immutable (git-based) Docker tag publishing

### PR DESCRIPTION
**Pull request by jackfrancis**
Based on https://github.com/deis/workflow/pull/487, and using a more cute name for the include `.mk` file just because!

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -47,6 +40,7 @@ test:
 # We also alter the RC file to set the image name.
 docker-build:
 	docker build --rm -t ${IMAGE} rootfs
+	docker tag -f ${IMAGE} ${MUTABLE_IMAGE}
 
 # Push to a registry that Kubernetes can access.
 docker-push:
```
Comment (sgoings) [inline Makefile]: You'll need to remove this docker-push target (it's in the versioning.mk file)

File: Makefile [hunk 2]
```diff
@@ -3,9 +3,7 @@ SHORT_NAME := workflow-manager
 # Enable vendor/ directory support.
 export GO15VENDOREXPERIMENT=1
 
-# SemVer with build information is defined in the SemVer 2 spec, but Docker
-# doesn't allow +, so we use -.
-VERSION := git-$(shell git rev-parse --short HEAD)
+include versioning.mk
```
Comment (sgoings) [inline Makefile]: I think you'll need to define `SHORT_NAME` above this include (see line 4-5 of versioning.mk).

Comment (jackfrancis) [inline Makefile]: SHORT_NAME is defined on line 1

Comment (sgoings) [inline Makefile]: Oops! It's there. GitHub was just bein' sneaky.

**Comment by sgoings** [comment]
Needs a rebase + squash, but then it LGTM.

**Pull request merged**